internal/manager_common.go

```diff
@@ -0,0 +1,28 @@
+package internal
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ConstructAtomPath adds the proper parameters for skip and top
+// This is common for the list operations for queues, topics and subscriptions.
+func ConstructAtomPath(basePath string, skip int, top int) string {
+	values := url.Values{}
+
+	if skip > 0 {
+		values.Add("$skip", fmt.Sprintf("%d", skip))
+	}
+
+	if top > 0 {
+		values.Add("$top", fmt.Sprintf("%d", top))
+	}
+
+	queryParams := values.Encode()
+
+	if len(queryParams) == 0 {
```

Comment:
Minor nit, I'd change this to `len(values)` before the call to `values.Encode()` to skip the encoding if there are no entries.